internal/provider/resource_ko_resolve.go

```diff
@@ -0,0 +1,263 @@
+package provider
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/google/ko/pkg/build"
+	"github.com/google/ko/pkg/commands"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"gopkg.in/yaml.v3"
+)
+
+func resolveConfig() *schema.Resource {
+	return &schema.Resource{
+		Description: "",
+
+		CreateContext: resourceKoResolveCreate,
+		ReadContext:   resourceKoResolveRead,
+		DeleteContext: resourceKoBuildDelete,
+
+		Schema: map[string]*schema.Schema{
+			FilenamesKey: {
+				Description: "Filenames, directorys, or URLs to files to use to create the resource",
+				Required:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+			},
+			RecursiveKey: {
+				Description: "Process the directory used in -f, --filename recursively. Useful when you want to manage related manifests organized within the same directory.",
+				Optional:    true,
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+			},
+			PushKey: {
+				Description: "Push images to KO_DOCKER_REPO",
+				Default:     true,
+				Optional:    true,
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+			},
+			SelectorKey: {
+				Description: "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			PlatformsKey: {
+				Description: "Which platform to use when pulling a multi-platform base. Format: all | <os>[/<arch>[/<variant>]][,platform]*",
+				Optional:    true,
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				ForceNew:    true,
+			},
+			SBOMKey: {
+				Description: "The SBOM media type to use (none will disable SBOM synthesis and upload, also supports: spdx, cyclonedx, go.version-m).",
+				Default:     "spdx",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
+			BaseImageKey: {
+				Description: "",
+				Default:     defaultBaseImage,
+				Optional:    true,
+				Type:        schema.TypeString,
+				ForceNew:    true,
+			},
```

Comment:
Is there any reason not to share these as much as possible with resource_ko_image?

Reply:
We should definitely share when it makes sense, I ~~took inspiration~~ copied from how terraform structures their official providers.

The vast majority of this stuff is boilerplate, so I chose the more verbose route instead of getting fancy with maximum reuse.